### PR TITLE
修正vs2019 vc142工具编译错误

### PR DESCRIPTION
**Pull request by xhawk18**
如题

### CODE REVIEW

**Comment by qicosmos** [comment]
1. use clang-format14 format code;
2. use pr template is help to understand what happend and why create the pr, provide more details is better;

File: include/struct_pack/struct_pack/struct_pack_impl.hpp [hunk 1]
```diff
@@ -1494,13 +1494,18 @@ class unpacker {
     return /*don't skip=*/false;
   }
 
+  template <int I, class... Ts>
+  decltype(auto) get_nth(Ts &&...ts) {
```
Comment (qicosmos) [inline include/struct_pack/struct_pack/struct_pack_impl.hpp:1498]: add ```STRUCT_PACK_INLINE```

**Comment by qicosmos** [comment]
could you update the pr again?

**Comment by xhawk18** [comment]
> could you update the pr again?

no,  
inline is not necessary here as described in C++ standard n3376 , 3.2/6, 

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf

**Comment by xhawk18** [comment]
> inline is not necessary here as described in C++ standard n4296, 3.2/6,

for example, std::make_shared()  has no inline in vc or gcc header files.
I personaly suggest to remove all unnecessary STRUCT_PACK_INLINE for more clear codes.

**Pull request merged**